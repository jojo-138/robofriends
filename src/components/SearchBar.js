export const SearchBar = ({ handleChange }) => {
    return (
        <div>
            <input 
                type="text" 
                placeholder="Search robofriends" 
                className="pa3 bn br3 bg-near-black white fw6 tracked shadow-5 mt3"
                onChange={handleChange}
            />
        </div>
    );
};