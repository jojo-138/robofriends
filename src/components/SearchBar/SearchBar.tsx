import { useAppDispatch, useAppSelector } from '../../hooks';
import { selectInput, setSearchField } from "../../features/setSearchField/searchField";

const SearchBar = () => {
    const dispatch = useAppDispatch();
    const input = useAppSelector(selectInput);
    
    return (
        <div>
            <input 
                type="text" 
                placeholder="Search robofriends" 
                className="pa3 bn br3 bg-near-black white fw6 tracked shadow-5 mt3"
                value={input}
                aria-label="search robofriends"
                onChange={(e) => dispatch(setSearchField(e.target.value))}
            />
        </div>
    );
};

export default SearchBar;
