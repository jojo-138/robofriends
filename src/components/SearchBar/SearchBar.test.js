import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import SearchBar from "./SearchBar";

describe('SearchBar component', () => {
  test('renders correctly', () => {
    expect(render(<SearchBar />)).toMatchSnapshot();
  })

  test('handleChange prop is being called when user types', () => {
    const mockHandleChange = jest.fn();
    render(<SearchBar handleChange={mockHandleChange} />);
    expect(mockHandleChange).not.toHaveBeenCalled();
    const input = screen.getByRole('textbox');
    userEvent.type(input, 'abc');
    expect(mockHandleChange).toHaveBeenCalledTimes(3);
  })
})