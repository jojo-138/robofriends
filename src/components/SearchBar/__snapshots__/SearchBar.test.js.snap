// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <input
          class="pa3 bn br3 bg-near-black white fw6 tracked shadow-5 mt3"
          placeholder="Search robofriends"
          type="text"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <input
        class="pa3 bn br3 bg-near-black white fw6 tracked shadow-5 mt3"
        placeholder="Search robofriends"
        type="text"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
